do.go

```diff
@@ -172,6 +172,16 @@ func (c *Client) do(ctx context.Context, body io.Reader, headers http.Header) (j
 	// decoding the response body for us, if it is encoded.
 	req.Header.Del("Accept-Encoding")
 
+	// Ensure headers compliance with GQL service expectations
+	if accept := headers.Get("Accept"); accept != "" {
```

Comment:
Can we optimise the code a bit?

Reply:
Following Anton suggestion in 1:1 chat, I am replacing that logic by simply always overriding the header by `application/json`. This package handles communication with GQL service and should only expect JSON as a response. (see new commit)